src/RouteJs/RouteJs.cs

```diff
@@ -67,13 +89,26 @@ private bool AllowRoute(RouteBase route)
 		/// <param name="route">The route</param>
 		/// <returns>Route information</returns>
 		private RouteInfo GetRoute(Route route)
-		{			
+		{
 			var routeInfo = new RouteInfo
 			{
-				Url = route.Url,
+				Url = GetUrl(route.Url),
 				Constraints = new RouteValueDictionary(),
 			};
 
+			if (IsLowerCaseUrls)
+			{
+				// Convert "controller" and "action" defaults to lowercase.
+				var convertedDefaults = route.Defaults
+					.Where(d => s_defaultKeysToConvert.Any(st => string.Equals(d.Key, st, StringComparison.OrdinalIgnoreCase)))
+					.Select(d => new KeyValuePair<string, object>(d.Key, ((string)d.Value).ToLower()))
+					.ToArray();
+				foreach (var item in convertedDefaults)
+				{
+					route.Defaults[item.Key] = item.Value;
```

Comment:
> this is not available to us in IDefaultsProcessor and we don't want global state

You can inject whatever you want into the class, as everything is using dependency injection. `RouteCollection` is already in the DI container (https://github.com/Daniel15/RouteJs/blob/master/src/RouteJs/ComponentRegistration.cs#L30) so you can just take the `RouteCollection` in the constructor and it'll work.